src/OpenLoco/src/GameCommands/RaiseLowerLand.cpp

```diff
@@ -0,0 +1,485 @@
+            return TileClearance::tileClearFunction(el, pos, removedBuildings, flags, totalCost);
+        };
+
+        QuarterTile qt(0xF, 0);
+        auto tileHeight = TileManager::getHeight(pos);
+        if (!TileClearance::applyClearAtAllHeights(pos, tileHeight.landHeight / 4, tileHeight.landHeight / 4, qt, clearFunc))
+        {
+            return GameCommands::FAILURE;
+        }
+
+        // Check bridge requirements for track and road elements
+        const auto tileIt = TileManager::get(pos);
+        for (auto& el : tileIt)
+        {
+            if (!(el.type() == ElementType::track || el.type() == ElementType::road))
+            {
+                continue;
+            }
+
+            if (el.isGhost())
+            {
+                continue;
+            }
+
+            auto height = el.baseZ() - targetBaseZ;
+            if (height < 0)
+            {
+                continue;
+            }
+
+            if (el.type() == ElementType::track)
+            {
+                auto* trackEl = el.as<TrackElement>();
+                if (trackEl != nullptr)
+                {
+                    if (trackEl->hasBridge())
+                    {
+                        auto* bridgeObj = ObjectManager::get<BridgeObject>(trackEl->bridge());
+                        if (height > bridgeObj->maxHeight)
+                        {
+                            setErrorText(StringIds::bridge_already_at_maximum_height);
+                            return FAILURE;
+                        }
+                    }
+
+                    auto* trackObj = ObjectManager::get<TrackObject>(trackEl->trackObjectId());
+                    FormatArguments args{};
+                    args.push(trackObj->name);
+                    setErrorText(StringIds::stringid_requires_a_bridge);
+                    return FAILURE;
+                }
+            }
+            else if (el.type() == ElementType::road)
+            {
+                auto* roadEl = el.as<RoadElement>();
+                if (roadEl != nullptr)
+                {
+                    if (roadEl->hasBridge())
+                    {
+                        auto* bridgeObj = ObjectManager::get<BridgeObject>(roadEl->bridge());
+                        if (height > bridgeObj->maxHeight)
+                        {
+                            setErrorText(StringIds::bridge_already_at_maximum_height);
+                            return FAILURE;
+                        }
+                    }
+
+                    auto* roadObj = ObjectManager::get<RoadObject>(roadEl->roadObjectId());
+                    FormatArguments args{};
+                    args.push(roadObj->name);
+                    setErrorText(StringIds::stringid_requires_a_bridge);
+                    return FAILURE;
+                }
+            }
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        surface = tileIt.surface();
+        if (isEditorMode())
```

Comment:
!isEditorMode we only do the clearing during normal mode.

Reply:
Oh, I thought it was countering the `setTerrainStyleAsCleared` (line 150).